lib/icingadb/icingadb-objects.cpp

```diff
@@ -161,17 +162,65 @@ void IcingaDB::UpdateAllConfigObjects()
 		std::vector<String> keys = GetTypeObjectKeys(lcType);
 		DeleteKeys(keys, Prio::Config);
 
-		auto objectChunks (ChunkObjects(type.first->GetObjects(), 500));
-
 		WorkQueue upqObjectType(25000, Configuration::Concurrency);
 		upqObjectType.SetName("IcingaDB:ConfigDump:" + lcType);
 
-		upqObjectType.ParallelFor(objectChunks, [this, &type, &lcType](decltype(objectChunks)::const_reference chunk) {
+		std::map<String, String> redisCheckSums;
+		String configCheckSum = m_PrefixConfigCheckSum + lcType;
+
+		upqObjectType.Enqueue([this, &configCheckSum, &redisCheckSums]() {
+			String cursor = "0";
+
+			do {
+				Array::Ptr res = m_Rcon->GetResultOfQuery({
+					"HSCAN", configCheckSum, cursor, "COUNT", "1000"
+				}, Prio::Config);
+
+				Array::Ptr kvs = res->Get(1);
+				Value* key = nullptr;
+				ObjectLock oLock (kvs);
+
+				for (auto& kv : kvs) {
+					if (key) {
+						redisCheckSums.emplace(std::move(*key), std::move(kv));
+						key = nullptr;
+					} else {
+						key = &kv;
+					}
+				}
+
+				cursor = res->Get(0);
+			} while (cursor != "0");
+		});
+
+		auto objectChunks (ChunkObjects(type.first->GetObjects(), 500));
+
+		std::map<String, std::vector<std::vector<String>>> ourContentRaw;
```

Comment:
Please add a descriptive comment what's stored in here, i.e. what do those strings represent.